src/index.js

```diff
@@ -58,6 +67,10 @@ module.exports = class Client {
       logger: this[PRIVATE.LOGGER],
       cluster,
       createPartitioner,
+      idempotent,
+      transactional,
+      transactionalId,
```

Comment:
I think it's reasonable to maintain a separate flag for transactional and transactionalId. This would allow a user to, for example, maintain a transactionalId value that it always present and a transactional flag that is conditionally switched. Also it's more explicit.

The idempotent flag cannot be false if the transactional flag is true. We can validate for this condition but it is odd. If we dislike the pattern than one route would be to expose a separate transaction producer & idempotent producer class or factory method.